src/pudl_archiver/archiver/classes.py

```diff
@@ -0,0 +1,165 @@
+"""Defines base class for archiver."""
+import asyncio
+import io
+import logging
+import tempfile
+import typing
+import zipfile
+from abc import ABC, abstractmethod
+from html.parser import HTMLParser
+from pathlib import Path
+
+import aiohttp
+
+from pudl_archiver.frictionless import ResourceInfo
+from pudl_archiver.zenodo.api_client import ZenodoDepositionInterface
+
+MEDIA_TYPES: dict[str, str] = {
+    "zip": "application/zip",
+    "xlsx": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+    "csv": "text/csv",
+}
+
+ArchiveAwaitable = typing.Generator[typing.Awaitable[tuple[Path, dict]], None, None]
```

Comment:
Can you include a docstring describing this type, please? Also, does it make sense to turn this into a class?

Reply:
I've added a docstring and used a NamedTuple to wrap the inner object. I'm not sure if there's a great way wrap the `Generator[Awaitable` thing though. It is a very confusing type to look at though

Reply:
Ah the named tuple feels clearer.  Thanks!